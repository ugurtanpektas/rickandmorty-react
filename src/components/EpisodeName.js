import React from "react";

class EpisodeDetail extends React.Component{

    constructor() {
        super();
        this.state = {episodeName: ""};
    }

    componentDidMount(){
        this.getEpisodeDetail();
    }
    
    getEpisodeDetail = async () => {
        const getEpisodeDetailCall = await fetch(this.props.episodeLink)
        const response = await getEpisodeDetailCall.json();
        this.setState({episodeName:response.name});
    }

    render(){
        return(
            <p>{this.state.episodeName}</p> 
        )    
    }
}

export default (EpisodeDetail);