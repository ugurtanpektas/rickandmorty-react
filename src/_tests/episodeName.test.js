import React from 'react';
import EpisodeName from '../components/EpisodeName';
import renderer from 'react-test-renderer';

test('<EpisodeName /> test passed', () => {
    const tree = renderer.create(
      <EpisodeName/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });