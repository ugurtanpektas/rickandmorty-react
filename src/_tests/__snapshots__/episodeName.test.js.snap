// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodeName /> test 1`] = `
<p>
  
</p>
`;

exports[`<EpisodeName /> test passed 1`] = `
<p>
  
</p>
`;
